sympy/polys/matrices/sdm.py

```diff
@@ -28,6 +28,37 @@ def __init__(self, elemsdict, shape, domain):
         if not all(0 <= c < n for row in self.values() for c in row):
             raise DDMBadInputError("Column out of range")
 
+    def getitem(self, i, j):
+        m, n = self.shape
+        if i >= m:
+                raise IndexError("row index out of range")
+        if j >= n:
+            raise IndexError("column index out of range")
+
+        try:
+            return self[i][j]
+        except KeyError:
+            return self.domain.zero
+
+    def extract_slice(self, slice1, slice2):
+        m, n = self.shape
+        ri = range(m)[slice1]
+        ci = range(n)[slice2]
+
+        sdm = {}
+        k = 0
+        for i, row in self.items():
+            if bool(row):
```

Comment:
Using `bool` does not do anything here.

Reply:
It handles the empty `dict` cases:

After commit:
```
In [1]: from sympy.polys.matrices.domainmatrix import DomainMatrix, DomainScalar, DDM,SDM

In [2]: M1 = DomainMatrix({0: {}, 1: {}, 2: {0: 1}}, (3,1), ZZ)

In [3]: M1[0:, 0]
Out[3]: DomainMatrix([[1]], (1, 1), ZZ)

In [4]: M1
Out[4]: DomainMatrix([[0], [0], [1]], (3, 1), ZZ)
```

Before commit:
```
In [2]: DomainMatrix({0: {}, 1: {0: 1}}, (2,1), ZZ)
Out[2]: DomainMatrix([[0], [1]], (2, 1), ZZ)

In [3]: DomainMatrix({0: {0: 0}, 1: {0: 1}}, (2,1), ZZ)
Out[3]: DomainMatrix([[0], [1]], (2, 1), ZZ)

In [4]: DomainMatrix({0: {}, 1: {0: 1}}, (2,2), ZZ)
Out[4]: DomainMatrix([[0, 0], [1, 0]], (2, 2), ZZ)
```

Reply:
The point is that `if row` is the same as `if bool(row)`.

Reply:
Sorry, I misunderstood the point.
Will remove `bool`.
Thanks.